integration/sawtooth_integration/tests/test_namespace_restriction.py

```diff
@@ -80,26 +95,62 @@ def make_batches(keys):
     return [imf.create_batch([('set', k, 0)]) for k in keys]
 
 
-class TestNamespacePermission(unittest.TestCase):
-    def test_namespace_permission(self):
-        """Tests that namespace permission onchain are letting block_info
-        transactions be injected and preventing intkey transactions to
-        be validated.
+def send_xo_cmd(cmd_str):
+    LOGGER.info('Sending xo cmd')
+    subprocess.run(
+        shlex.split(cmd_str),
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        check=True)
+
+
+class TestNamespaceRestriction(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        wait_for_rest_apis(['rest-api:8008'])
+
+    def test_namespace_restriction(self):
         """
-        batches = make_batches('abcd')
+        Tests that namespaces stored on-chain are enforced by the
+        validators. According to the sawtooth_settings declared in the docker
+        compose file, the transaciton families are expected to behave
+        as follows:
+        - block_info transactions are allowed
+        - intkey transactions are banned
+        - xo transactions are allowed
+        """
+        batches = make_batches('abcdef')
+
+        send_xo_cmd('sawtooth keygen')
+
+        xo_cmds = [
+            'xo create game',
+            'xo take game 5',
+            'xo take game 5',
+            'xo take game 9',
+            'xo create game',
+            'xo take game 4',
+        ]
 
         # Assert all block info transactions are committed
         for i, batch in enumerate(batches):
             post_batch(batch)
+            send_xo_cmd('{} --url {} --wait {}'.format(
```

Comment:
Does this remove the need for the sleep below?  This should probably be in a loop, as the REST API may not wait the total WAIT amount requested.

Reply:
good catch! It makes sense